cmd/snap/inhibit.go

```diff
@@ -145,95 +143,57 @@ type inhibitionFlow interface {
 	FinishInhibitionNotification(ctx context.Context) error
 }
 
-var newInhibitionFlow = func(instanceName string) inhibitionFlow {
-	if isGraphicalSession() {
-		return &graphicalFlow{instanceName: instanceName}
-	}
-	return &textFlow{instanceName: instanceName}
-}
-
-type textFlow struct {
-	instanceName string
-}
-
-func (tf *textFlow) StartInhibitionNotification(ctx context.Context) error {
-	_, err := fmt.Fprintf(Stdout, i18n.G("snap package %q is being refreshed, please wait\n"), tf.instanceName)
-	// TODO: add proper progress spinner
-	return err
-}
-
-func (tf *textFlow) FinishInhibitionNotification(ctx context.Context) error {
-	return nil
+var newInhibitionFlow = func(cli *client.Client, instanceName string) inhibitionFlow {
+	return &noticesFlow{instanceName: instanceName, cli: cli}
 }
 
-type graphicalFlow struct {
+type noticesFlow struct {
 	instanceName string
 
-	notifiedDesktopIntegration bool
+	cli *client.Client
 }
 
-func (gf *graphicalFlow) StartInhibitionNotification(ctx context.Context) error {
-	gf.notifiedDesktopIntegration = tryNotifyRefreshViaSnapDesktopIntegrationFlow(ctx, gf.instanceName)
-	if gf.notifiedDesktopIntegration {
-		return nil
+func (gf *noticesFlow) StartInhibitionNotification(ctx context.Context) error {
+	opts := client.NotifyOptions{
+		Type: client.SnapRunInhibitNotice,
+		Key:  gf.instanceName,
 	}
-
-	// unable to use snapd-desktop-integration, let's fall back to graphical session flow
-	refreshInfo := client.PendingSnapRefreshInfo{
-		InstanceName: gf.instanceName,
-		// remaining time = 0 results in "Snap .. is refreshing now" message from
-		// usersession agent.
-		TimeRemaining: 0,
+	_, err := gf.cli.Notify(&opts)
+	if err != nil {
+		return err
 	}
-	return pendingRefreshNotification(ctx, &refreshInfo)
-}
 
-func (gf *graphicalFlow) FinishInhibitionNotification(ctx context.Context) error {
-	if gf.notifiedDesktopIntegration {
-		// snapd-desktop-integration detects inhibit unlock itself, do nothing
-		return nil
+	// Fallback to text notification if marker "snap-refresh-observe"
+	// interface is not connected.
+	if isStdoutTTY && !markerInterfaceConnected(gf.cli) {
+		fmt.Fprintf(Stdout, i18n.G("snap package %q is being refreshed, please wait\n"), gf.instanceName)
```

Comment:
are there cases where regardless of sitting on a tty we'd like to produce the message to stderr? TBH even with the current check's I'd still write this to stderr

Reply:
Good point, I will update it.

Reply:
updated, thank you!